Doc/library/stdtypes.rst

```diff
@@ -1369,6 +1369,49 @@ Lists are mutable sequences, typically used to store collections of
 homogeneous items (where the precise degree of similarity will vary by
 application).
 
+.. note::
+
+   Individual operations on :class:`list` instances such as
+   :meth:`~list.append`, :meth:`~list.pop`, ``lst[i] = x``, and ``x = lst[i]``
+   are atomic and will not corrupt the list or crash when called concurrently
+   from multiple threads:
+
+   .. code-block::
+      :class: good
+
+      # These operations are safe to call concurrently from multiple threads
+      lst.append(item)       # atomically appends item
+      lst.pop()              # atomically removes and returns last item
+      lst[i] = value         # atomically replaces item at index i
+      item = lst[i]          # atomically retrieves item at index i
+
+   One exception to this rule is :meth:`~list.extend`. Its atomicity
```

Comment:
What about `sort`? Does the thread-safety rely on the CPython implementation detail?

Reply:
`sort` uses a critical section so it is atomic. `index` and `count` do not use critical section around the whole operation, they just use `getref` APIs. That means that if a list is concurrently mutated, the index/count might change.

`min`/`max` create iterators which isn't thread-safe. But I think that should be part of `min`/`max`'s documentation rather than be mentioned here.